src/lib/storage/mvcc_data.hpp

```diff
@@ -36,9 +36,9 @@ struct MvccData {
   /**
    * Grows mvcc data by the given delta
    *
-   * @param begin_cid value all new begin_cids will be set to
+   * @param transaction_id    value all new tids will be set to
    */
-  void grow_by(size_t delta, CommitID begin_cid);
+  void grow_by(size_t delta, TransactionID transaction_id);
```

Comment:
I don't think so, the issue was a race of two Insert::_on_execute() running at the same time, both allocating space and then trying to insert **until the table was "full"** instead of stopping when all their input was consumed.